File: sdk-node/examples/subscriptions/retrieve.ts
/**
 * Retrieve a single subscription by ID.
 *
 * Run:
 *   npx tsx examples/subscriptions/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const sub = await client.subscriptions.retrieve('sub_replace_with_real_id')

console.log(`subscription ${sub.id ?? '?'} [${sub.status ?? '?'}]`)
console.log(`  price          ${sub.priceId ?? '?'} × ${String(sub.quantity ?? 0)}`)
console.log(`  current period ${sub.currentPeriodStart ?? '?'} → ${sub.currentPeriodEnd ?? '?'}`)
console.log(`  cancelAtPeriodEnd: ${String(sub.cancelAtPeriodEnd ?? false)}`)
console.log(`  autoCharge: ${String(sub.autoCharge ?? false)}`)

File: sdk-python/examples/subscriptions/retrieve.py
"""Run with: python examples/subscriptions/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        sub = client.subscriptions.retrieve("sub_replace_with_real_id")
        print(f"subscription {sub.id} [{sub.status}]")
        print(f"  price          {sub.price_id} x {sub.quantity}")
        print(f"  current period {sub.current_period_start} → {sub.current_period_end}")
        print(f"  cancelAtPeriodEnd: {sub.cancel_at_period_end}")
        print(f"  autoCharge: {sub.auto_charge}")


if __name__ == "__main__":
    main()
